brainiak/matnormal/regression.py

```diff
@@ -0,0 +1,155 @@
+import tensorflow as tf
+import numpy as np
+from sklearn.base import BaseEstimator
+from brainiak.matnormal.matnormal_likelihoods import matnorm_logp
+from tensorflow.contrib.opt import ScipyOptimizerInterface
+
+__all__ = ['MatnormRegression']
+
+
+class MatnormRegression(BaseEstimator):
+    """ This analysis allows maximum likelihood estimation of regression models
+    in the presence of both spatial and temporal covariance.
+
+    ..math::
+    Y \\sim \\mathcal{MN}(X\beta, time_cov, space_cov)
+
+    Parameters
+    ----------
+    time_cov : subclass of CovBase
+        TR noise covariance class following CovBase interface.
+    space_cov : subclass of CovBase
+        Voxel noise covariance class following CovBase interface.
+    learnRate : real, default=0.01
+        Step size for the Adam optimizer
+
+    """
+    def __init__(self, time_cov, space_cov,
+                 optimizer='L-BFGS-B', optCtrl=None):
+
+        self.optCtrl, self.optMethod = optCtrl, optimizer
+        self.time_cov = time_cov
+        self.space_cov = space_cov
+
+        self.n_t = time_cov.size
+        self.n_v = space_cov.size
+
+        self.Y = tf.placeholder(tf.float64, [self.n_t, self.n_v], name="Y")
+
+        self.X = tf.placeholder(tf.float64, [self.n_t, None], name="X")
+
+        # create a tf session we reuse for this object
+        self.sess = tf.Session()
+
+    # @define_scope
+    def logp(self):
+        """ Log likelihood of model (internal)
+        """
+        y_hat = tf.matmul(self.X, self.beta)
+        resid = self.Y - y_hat
+        return matnorm_logp(resid, self.time_cov, self.space_cov)
+
+    def fit(self, X, y):
```

Comment:
The optional arguments in the docstring (e.g. voxel_pos, times) are not inputs in the def statement. Is this by design? (Is CovGP not implemented?)

Reply:
yeah, outdated docstring (covgp was excluded from this PR when I split things up from the private repo)